### PR TITLE
Support for java 17 in generated projects - Codestarts 

### LINKED ISSUE #20752: Java 17 is not reflected as source/target of generated project
### Describe the bug

Java 17 is not reflected as source/target of generated project.

In dev mode one sees warning like this `2021-10-13 22:56:53,445 WARN  [io.qua.dep.dev.JavaCompilationProvider] (vert.x-worker-thread-0) system modules path not set in conjunction with -source 11, line -1 in [unknown source]`

### Expected behavior

Project should use 17 as the value for maven.compiler. source / target / (release?).

### Actual behavior

Generated pom.xml when using Java 17 looks like this:

```xml
    <maven.compiler.source>11</maven.compiler.source>
    <maven.compiler.target>11</maven.compiler.target>
```

### How to Reproduce?

setJdk17

mvn io.quarkus.platform:quarkus-maven-plugin:2.4.0.CR1:create -DprojectGroupId=my-groupId -DprojectArtifactId=app-generated-skeleton -DprojectVersion=1.0.0-SNAPSHOT -DpackageName=org.my.group -DquarkusRegistryClient=false -DplatformArtifactId=quarkus-bom -DplatformVersion=2.4.0.CR1
mvn -f app-generated-skeleton quarkus:dev

sed -i "" "s/RESTEasy/RESTEasy v2/" app-generated-skeleton/src/main/java/org/my/group/GreetingResource.java
curl localhost:8080/hello

check the log, you will see "WARN  [io.qua.dep.dev.JavaCompilationProvider] (vert.x-worker-thread-0) system modules path not set in conjunction with -source 11, line -1 in [unknown source]"

### Output of `uname -a` or `ver`

macOS BigSur

### Output of `java -version`

Java 17

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.4.0.CR1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by netodevel**
Fixes #20752

### CODE REVIEW

**Comment by ia3andy** [comment]
It looks pretty good, don't forget to squash it to 1 commit

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/src/main/docker/Dockerfile.tpl.qute.jvm [hunk 1]
```diff
@@ -2,7 +2,7 @@
     {#quarkusbuild}{buildtool.cli} {buildtool.cmd.package}{/quarkusbuild}
     {#image}{dockerfile.jvm.from} {/image}
     {#args}
-ARG JAVA_PACKAGE={dockerfile.jvm.java-package}
```
[the remark this answers is not in the document]
Comment (Chexpir) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/src/main/docker/Dockerfile.tpl.qute.jvm:5]: Hi! any news on the red hat repo side? I cannot comment on google groups.

**Comment by ia3andy** [comment]
@netodevel the JDK 17 is now available. Instead of installing the package, we are going to use a different base image (as suggested by @cescoffier on zulip):
```
####
# This Dockerfile is used in order to build a container that runs the Quarkus application in JVM mode
#
# Before building the container image run:
#
# ./mvnw package
#
# Then, build the image with:
#
# docker build -f src/main/docker/Dockerfile.jvm -t quarkus/getting-started-jvm .
#
# Then run the container using:
#
# docker run -i --rm -p 8080:8080 quarkus/getting-started-jvm
#
# If you want to include the debug port into your docker image
# you will have to expose the debug port (default 5005) like this :  EXPOSE 8080 5050
#
# Then run the container using :
#
# docker run -i --rm -p 8080:8080 -p 5005:5005 -e JAVA_ENABLE_DEBUG="true" quarkus/getting-started-jvm
#
###
FROM registry.redhat.io/ubi8/openjdk-17-runtime:1.10-5

ARG RUN_JAVA_VERSION=1.3.8
ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en'

# Configure the JAVA_OPTIONS, you can add -XshowSettings:vm to also display the heap size.
ENV JAVA_OPTIONS="-Dquarkus.http.host=0.0.0.0 -Djava.util.logging.manager=org.jboss.logmanager.LogManager"
# We make four distinct layers so if there are application changes the library layers can be re-used
COPY --chown=185 target/quarkus-app/lib/ /deployments/lib/
COPY --chown=185 target/quarkus-app/*.jar /deployments/
COPY --chown=185 target/quarkus-app/app/ /deployments/app/
COPY --chown=185 target/quarkus-app/quarkus/ /deployments/quarkus/

EXPOSE 8080
USER 185

ENTRYPOINT [ "java", "-jar", "/deployments/quarkus-run.jar" ]
```

We need to also use a openjdk runtime base image for previous versions.

As this is a priority, if you are not available, I will continue the dev for this later today (but you'll keep the credit for your work of course), let me know..

**Comment by ia3andy** [comment]
I prepared a commit with the changes: https://github.com/ia3andy/quarkus/commit/fdf6124d7c2ae4a96646b51e1b02a8407a0e3454

But it seems that jdk 17 is on a different repository `registry.redhat.io` which has an extra layer of security:
https://catalog.redhat.com/software/containers/ubi8/openjdk-17-runtime/618bdc5f843af1624c4e4ba8?container-tabs=gti&gti-tabs=registry-tokens

IMO having to login or use a token to pull the image is a blocker.

cc @maxandersen @cescoffier

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute [hunk 2]
```diff
@@ -18,31 +18,19 @@
...
 #
-# docker run -i --rm -p 8080:8080 -p 5005:5005 -e JAVA_ENABLE_DEBUG="true" quarkus/{project.artifact-id}-{type}
+# docker run -i --rm -p 8080:8080 quarkus/{project.artifact-id}-{type}
 #
 ###
-FROM {#insert image /}
-{#insert args /}
+FROM registry.access.redhat.com/ubi8/openjdk-{java.version}-runtime:1.10
+
+ARG RUN_JAVA_VERSION={dockerfile.run-java-version}
```
Comment (gastaldi) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute:26]: This is no longer needed

**Comment by ia3andy** [comment]
I pushed my commit, this PR needs to be squashed and rebase from `main`.

Java 17 Runtime is in the process of being pushed to the unauthenticated repository:
https://issues.redhat.com/browse/CLOUDDST-10475

This PR will have to wait for this registry.access.redhat.com/ubi8/openjdk-17-runtime to be available.

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute (hunk 2 above)
Comment (ia3andy) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute]: Fixed

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute [hunk 3]
```diff
@@ -18,31 +18,18 @@
...
-    && chmod 540 /deployments/run-java.sh \
-    && echo "securerandom.source=file:/dev/urandom" >> /etc/alternatives/jre/conf/security/java.security
 
 # Configure the JAVA_OPTIONS, you can add -XshowSettings:vm to also display the heap size.
 ENV JAVA_OPTIONS="-Dquarkus.http.host=0.0.0.0 -Djava.util.logging.manager=org.jboss.logmanager.LogManager"
 {#insert copy /}
 
 EXPOSE 8080
-USER 1001
+USER 185
```
Comment (gastaldi) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute:33]: Do we still need to set the user here?

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java [hunk 4]
```diff
@@ -78,10 +82,16 @@ public static void setJavaVersion(Map<String, Object> values, String javaTarget)
...
-        if (matcher.matches() && Integer.parseInt(matcher.group(1)) < 11) {
-            values.put(ProjectGenerator.JAVA_TARGET, "8");
+
+        if (matcher.matches()) {
+            int versionExtracted = Integer.parseInt(matcher.group(1));
+            int version = JAVA_VERSIONS_LTS.stream()
+                    .filter(e -> e.equals(versionExtracted))
+                    .findFirst().orElse(DEFAULT_JAVA_VERSION);
+
+            values.put(ProjectGenerator.JAVA_TARGET, String.valueOf(version));
```
Comment (gastaldi) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java:92]: In this case it's easier to deal with `String` directly instead of `Integer` (avoids parsing to int and formatting back to String)

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute (hunk 3 above)
Comment (ia3andy) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute:33]: @cescoffier ?

**Comment by ia3andy** [comment]
@netodevel you need to run `mvn process-sources` to format the files and fix the build.

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute (hunk 3 above)
Comment (cescoffier) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute:33]: Yes, it's all about security (and openshift guidelines)

**Comment by ia3andy** [comment]
It seems you need to fix the tests snapshot files (by running them with `-Dsnap`)

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute (hunk 3 above)
Comment (ebullient) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute:33]: Will the switch from 1001 cause issues?

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/src/main/docker/Dockerfile.tpl.qute.legacy-jar [hunk 5]
```diff
@@ -1,12 +1,7 @@
...
     {#quarkusbuild}{buildtool.cli} {buildtool.cmd.package-legacy-jar}{/quarkusbuild}
-    {#image}{dockerfile.legacy-jar.from} {/image}
-    {#args}
-ARG JAVA_PACKAGE={dockerfile.legacy-jar.java-package}
-ARG RUN_JAVA_VERSION={dockerfile.legacy-jar.run-java-version}
-    {/args}
     {#copy}
 COPY {buildtool.build-dir}/lib/* /deployments/lib/
-COPY {buildtool.build-dir}/*-runner.jar /deployments/app.jar
+COPY {buildtool.build-dir}/*-runner.jar /deployments/quarkus-run.jar
```
Comment (ebullient) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/src/main/docker/Dockerfile.tpl.qute.legacy-jar:5]: Is this change necessary or just preferred?

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java (hunk 4 above)
Comment (ebullient) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java:92]: do we not need the “anything less than 11 means 8” logic anymore?

Comment (ebullient) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java:92]: (do we support creating projects for 8 anymore? As of Quarkus 2, we’ve moved on to Java 11… is that the floor?)

Comment (netodevel) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java]: @gastaldi @ebullient,thanks for the appointments
Fixs done.

Comment (gastaldi) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java]: > (do we support creating projects for 8 anymore? As of Quarkus 2, we’ve moved on to Java 11… is that the floor?)

Yes, Java 11 is the minimum. Java 8 is not supported anymore

Comment (netodevel) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java]: i removed the possibility to set java 8

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/src/main/docker/Dockerfile.tpl.qute.legacy-jar (hunk 5 above)
Comment (netodevel) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/src/main/docker/Dockerfile.tpl.qute.legacy-jar:5]: I believe it doesn't matter it's just a matter of the jar's naming.

@ia3andy, do you have any consideration?

**Comment by netodevel** [comment]
@ia3andy, about the tests it was necessary to adjust some assertions and add some new ones.
I don't know what `-Dsnap` does.

File: independent-projects/tools/devtools-testing/src/test/java/io/quarkus/devtools/codestarts/quarkus/QuarkusCodestartGenerationTest.java [hunk 6]
```diff
@@ -276,15 +276,15 @@ private void checkDockerfilesWithMaven(Path projectDir) {
         assertThat(projectDir.resolve("src/main/docker/Dockerfile.jvm")).exists()
                 .satisfies(checkContains("./mvnw package"))
                 .satisfies(checkContains("docker build -f src/main/docker/Dockerfile.jvm"))
-                .satisfies(checkContains("registry.access.redhat.com/ubi8/ubi-minimal:8.4"))
```
Comment (netodevel) [inline independent-projects/tools/devtools-testing/src/test/java/io/quarkus/devtools/codestarts/quarkus/QuarkusCodestartGenerationTest.java:279]: @ia3andy, about the tests it was necessary to adjust some assertions and add some new ones.
I don't know what `-Dsnap` does.

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java [hunk 7]
```diff
@@ -78,10 +82,15 @@ public static void setJavaVersion(Map<String, Object> values, String javaTarget)
...
         Matcher matcher = JAVA_VERSION_PATTERN
                 .matcher(javaTarget != null ? javaTarget : System.getProperty("java.version", ""));
-        if (matcher.matches() && Integer.parseInt(matcher.group(1)) < 11) {
-            values.put(ProjectGenerator.JAVA_TARGET, "8");
+
+        if (matcher.matches()) {
+            String versionExtracted = matcher.group(1);
+            String version = JAVA_VERSIONS_LTS.stream()
+                    .filter(e -> e.equals(versionExtracted))
+                    .findFirst().orElse(DEFAULT_JAVA_VERSION);
```
Comment (gastaldi) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java:90]:
```suggestion
            String version = JAVA_VERSIONS_LTS.contains(versionExtracted) ? versionExtracted : DEFAULT_JAVA_VERSION;
```

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/src/main/docker/Dockerfile.tpl.qute.legacy-jar (hunk 5 above)
Comment (ia3andy) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/src/main/docker/Dockerfile.tpl.qute.legacy-jar:5]: Yes this is needed by the common entrypoint. The alternative would be to pass a parameter to deal with names in the entrypoint (but that's more complex to achieve and less clean IMO):
```
ENTRYPOINT [ "java", "-jar", "/deployments/quarkus-run.jar" ]
```

File: independent-projects/tools/devtools-testing/src/test/java/io/quarkus/devtools/codestarts/quarkus/QuarkusCodestartGenerationTest.java (hunk 6 above)
Comment (ia3andy) [reply independent-projects/tools/devtools-testing/src/test/java/io/quarkus/devtools/codestarts/quarkus/QuarkusCodestartGenerationTest.java:279]: @netodevel => https://quarkus.io/guides/extension-codestart#snapshot-testing

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java (hunk 7 above)
Comment (ia3andy) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java:90]: Yes, using contains is more efficient. Thanks @gastaldi

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/buildtool/gradle-kotlin-dsl/base/build-layout.include.qute [hunk 8]
```diff
@@ -53,6 +53,9 @@ java {
     {#if java.version == "11"}
     sourceCompatibility = JavaVersion.VERSION_11
     targetCompatibility = JavaVersion.VERSION_11
+    {#else if java.version == "17"}
+    sourceCompatibility = JavaVersion.VERSION_17
+    targetCompatibility = JavaVersion.VERSION_17
     {#else}
     sourceCompatibility = JavaVersion.VERSION_1_8
     targetCompatibility = JavaVersion.VERSION_1_8
```
Comment (ia3andy) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/buildtool/gradle-kotlin-dsl/base/build-layout.include.qute:61]:
```suggestion
    {#if java.version == "17"}
    sourceCompatibility = JavaVersion.VERSION_17
    targetCompatibility = JavaVersion.VERSION_17
    {#else}
    sourceCompatibility = JavaVersion.VERSION_11
    targetCompatibility = JavaVersion.VERSION_11
```

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/buildtool/gradle/base/build-layout.include.qute [hunk 9]
```diff
@@ -49,6 +49,9 @@ java {
     {#if java.version == "11"}
     sourceCompatibility = JavaVersion.VERSION_11
     targetCompatibility = JavaVersion.VERSION_11
+    {#else if java.version == "17"}
+    sourceCompatibility = JavaVersion.VERSION_17
+    targetCompatibility = JavaVersion.VERSION_17
     {#else}
     sourceCompatibility = JavaVersion.VERSION_1_8
     targetCompatibility = JavaVersion.VERSION_1_8
```
Comment (ia3andy) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/buildtool/gradle/base/build-layout.include.qute:57]:
```suggestion
    {#if java.version == "17"}
    sourceCompatibility = JavaVersion.VERSION_17
    targetCompatibility = JavaVersion.VERSION_17
    {#else}
    sourceCompatibility = JavaVersion.VERSION_11
    targetCompatibility = JavaVersion.VERSION_11
```

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java [hunk 10]
```diff
@@ -18,6 +19,9 @@
 import javax.lang.model.SourceVersion;
 
 public class CreateProjectHelper {
+
+    private static final List<String> JAVA_VERSIONS_LTS = List.of("11", "17");
```
Comment (ia3andy) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java:23]:
```suggestion
    private static final Set<String> JAVA_VERSIONS_LTS = Set.of("11", "17");
```

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java [hunk 11]
```diff
@@ -8,6 +8,7 @@
 import java.nio.file.Path;
 import java.util.Collection;
 import java.util.HashSet;
+import java.util.List;
```
Comment (ia3andy) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java:11]:
```suggestion
import java.util.Set;
```

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java (hunk 7 above)
Comment (ia3andy) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java:90]: I also added a suggestion to have a `Set`

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/buildtool/gradle-kotlin-dsl/base/build-layout.include.qute (hunk 8 above)
Comment (netodevel) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/buildtool/gradle-kotlin-dsl/base/build-layout.include.qute:61]: @ia3andy, wow, good point
I'll fix it as soon as I have some time.

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java (hunk 7 above)
Comment (netodevel) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java]: @gastaldi @ia3andy, done!!

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/buildtool/gradle-kotlin-dsl/base/build-layout.include.qute (hunk 8 above)
Comment (netodevel) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/buildtool/gradle-kotlin-dsl/base/build-layout.include.qute]: @ia3andy, done.

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java (hunk 10 above)
Comment (netodevel) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java]: Done.

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java (hunk 11 above)
Comment (netodevel) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/codegen/CreateProjectHelper.java]: Done.

**Review by gastaldi — APPROVED** [review]
Looks good to me! Great job!

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute [hunk 12]
```diff
@@ -18,31 +18,18 @@
 #
 # Then run the container using :
 #
-# docker run -i --rm -p 8080:8080 -p 5005:5005 -e JAVA_ENABLE_DEBUG="true" quarkus/{project.artifact-id}-{type}
+# docker run -i --rm -p 8080:8080 quarkus/{project.artifact-id}-{type}
 #
 ###
-FROM {#insert image /}
-{#insert args /}
+FROM registry.access.redhat.com/ubi8/openjdk-{java.version}-runtime:1.10
```
Comment (maxandersen) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute:24]: I thought I put this comment but can't find it. Does this not break former cli running these as they expect image to be passed in ?

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute (hunk 3 above)
Comment (maxandersen) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute:33]: I have same question as ebullient ?

Comment (ebullient) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute:33]: Right -- I think we're flipping over to always use the pre-baked UBI images... which is why the jar file name has to change, and why the user id has to change...

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute (hunk 12 above)
Comment (netodevel) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute:24]: @maxandersen, I'm not contextualized with this cli.
which cli are you talking about? and which command would impact?

Comment (ia3andy) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute:24]: We are safe, each codestart is kept as a unit.

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute (hunk 3 above)
Comment (ia3andy) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute:33]: yes, exactly.

**Review by ia3andy — APPROVED** [review]
I think we are good to go.

Thanks a lot @netodevel!!! This was not an easy one to take to completion 🍾🥳

**Comment by ia3andy** [comment]
If we have problems we can always iterate from this..

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute (hunk 12 above)
Comment (maxandersen) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute:24]: So if using quarkus 2.6 cli or mvn plugin how does it know how to pass in "image" to the former version ?

Is this defined somewhere else ?

Comment (ia3andy) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/Dockerfile-layout.include.qute:24]: The image is part of the template so no need for data. The `{java.version}` already exists in previous versions

**Comment by netodevel** [comment]
@ia3andy, Thanks a lot for all the help you gave me :)

**Comment by geoand** [comment]
IIUC, the base image has been changed, correct? If so, this is a breaking change and we should label it as such.
